crates/turborepo-lib/src/config/mod.rs

```
@@ -0,0 +1,651 @@
+        with_allow_no_package_manager,
+        allow_no_package_manager,
+        Option<bool>
+    );
+    create_builder!(with_daemon, daemon, Option<bool>);
+    create_builder!(with_env_mode, env_mode, Option<EnvMode>);
+    create_builder!(with_cache_dir, cache_dir, Option<Utf8PathBuf>);
+    create_builder!(
+        with_root_turbo_json_path,
+        root_turbo_json_path,
+        Option<AbsoluteSystemPathBuf>
+    );
+
+    pub fn build(&self) -> Result<ConfigurationOptions, Error> {
+        // Priority, from least significant to most significant:
+        // - shared configuration (turbo.json)
+        // - global configuration (~/.turbo/config.json)
+        // - local configuration (<REPO_ROOT>/.turbo/config.json)
+        // - environment variables
+        // - CLI arguments
+        // - builder pattern overrides.
+
+        let turbo_json = TurboJsonReader::new(&self.repo_root);
+        let global_config = ConfigFile::global_config(self.global_config_path.clone())?;
+        let global_auth = AuthFile::global_auth(self.global_config_path.clone())?;
+        let local_config = ConfigFile::local_config(&self.repo_root);
+        let env_vars = self.get_environment();
+        let env_var_config = EnvVars::new(&env_vars)?;
+        let override_env_var_config = OverrideEnvVars::new(&env_vars)?;
+
+        // These are ordered from highest to lowest priority
+        let sources: [Box<dyn ResolvedConfigurationOptions>; 7] = [
+            Box::new(override_env_var_config),
+            Box::new(&self.override_config),
+            Box::new(env_var_config),
+            Box::new(local_config),
+            Box::new(global_auth),
+            Box::new(global_config),
+            Box::new(turbo_json),
+        ];
+
+        let config = sources.into_iter().try_fold(
+            ConfigurationOptions::default(),
+            |mut acc, current_source| {
+                let mut current_source_config = current_source.get_configuration_options(&acc)?;
```

There was an existing bug where we would ignore any errors that came from resolving config options from a source. In reality since these were (mostly) eagerly calculated it only affected and validation errors from extracting config options from a `turbo.json`.

We now bubble up any errors generated by resolving config options.